php/libraries/Database.class.inc

```diff
@@ -744,6 +744,77 @@ class Database
         return $result;
     }
 
+    /**
+     * Runs an SQL select statement as a prepared query and re-indexes
+     * the results using the given primary key.
```

Comment:
It looks like it needs to be a unique non-null column, not necessarily a primary key (which also affects the function name..)

Reply:
yeah your correct I had primary key everywhere before then changed it all except here I guess